### PR TITLE
Update kafka tests to comply with new structure

### PR DESCRIPTION
**Pull request by athos-ribeiro**
The kafka image was split into a kafka and a zookeeper image. Here, we
adapt the tests to the new images structure.

Signed-off-by: Athos Ribeiro <athos.ribeiro@canonical.com>

### CODE REVIEW

**Comment by sergiodj** [comment]
Thanks, Athos.

Running the tests locally (using the images published on Dockerhub) gives me the following:

```
test_local_port_binding
test_communication
test_producer_consumer
ASSERT:Could not consume message. Was the message produced?
test_manifest_exists

Ran 4 tests.

FAILED (failures=1)
```

I remember running these tests before the split and they were passing, so maybe the error was introduced after the split?

Ah, a small nit: could you also prefix the commit message with `OCI: `, please?

File: oci-unit-tests/kafka_test.sh [hunk 1]
```diff
@@ -53,8 +54,7 @@ docker_run_zookeeper() {
      -d \
      -p "${ZOOKEEPER_PORT}":"${ZOOKEEPER_PORT}" \
      --name "${DOCKER_PREFIX}"_zookeeper_"${id}" \
-     "${DOCKER_IMAGE}" \
-     zookeeper-server-start.sh /etc/kafka/zookeeper.properties
+     "${ZOOKEEPER_IMAGE}"
 }
 
 # Helper function to execute kafka with some common arguments.
```
Comment (sergiodj) [inline oci-unit-tests/kafka_test.sh:60]: This comment needs to be updated.

**Review by sergiodj — CHANGES_REQUESTED** [review]
Aside from a small request to update a comment, everything else LGTM.  I'm marking this review as "Request changes" though, because one of the tests is failing locally.

**Comment by athos-ribeiro** [comment]
Thanks, Sergio.

The tests are not failing here. Any change you were using the old images locally?

I updated the commit message as requested. I am not sure if I understood the request for updating the comment for running the kafka image though....

**Comment by sergiodj** [comment]
> The tests are not failing here. Any change you were using the old images locally?

Sigh, it seems like it.  I removed the kafka and zookeeper images and ran the test again, and now it passes.  But that's strange; the `oneTimeSetUp` function is calling `remove_current_image`, which should have taken care of refreshing the kafka image for me (it doesn't refresh the zookeeper image, but I didn't have it the first time I ran the test anyway).  Hmm...

> I updated the commit message as requested. I am not sure if I understood the request for updating the comment for running the kafka image though....

Thanks.  The request to update the comment was based on a misunderstanding: I thought that `docker_run_broker` was running something else other than kafka, but it turns out that it's actually running kafka itself.  I'm OK with keeping this function name, but I would personally have named the function `docker_run_kafka`, I think.  Anyway, not a big deal.

Thanks, +1 from me.

**Review by sergiodj — APPROVED** [review]

**Pull request merged**